app/controllers/Application.scala

```diff
@@ -29,8 +32,9 @@ class Application(
     payPalConfigProvider: PayPalConfigProvider,
     paymentAPIService: PaymentAPIService,
     stringsConfig: StringsConfig,
-    settingsProvider: SettingsProvider
-)(implicit val ec: ExecutionContext) extends AbstractController(components) with SettingsSurrogateKeySyntax {
+    settingsProvider: SettingsProvider,
+    guardianDomain: GuardianDomain
+)(implicit val ec: ExecutionContext) extends AbstractController(components) with SettingsSurrogateKeySyntax with StrictLogging with ServersideAbTestCookie {
```

Comment:
`with ServersideAbTestCookie` I'm not keen on using traits to import things without a namespace, you can get exactly the same concept just using an object but without these extra links
```scala
object WithServersideAbTestCookie {
  def apply(blahblah) = {
    .....
  }
}
```

Reply:
Yeah, again this is how `.withSettingsSurrogateKey` works so I may switch them both over to functions in a future PR. Interested if @Mullefa has any additional arguments in favour of the trait + implicit class style here since he introduced it.

Reply:
agree Guy, some things are conceptually different and some things are just two ways to write the same thing.  Furthermore, some things actually overspecify the relationships between different parts of the code.
I think we should be harshest against the third case, but I agree we should try to improve matters in the middle case, as that is a more human readability parsing issue

Reply:
nice way of putting it John!